prov/util/src/util_attr.c

```diff
@@ -1096,6 +1096,8 @@ static void fi_alter_ep_attr(struct fi_ep_attr *attr,
 		attr->tx_ctx_cnt = hints->tx_ctx_cnt;
 	if (hints->rx_ctx_cnt)
 		attr->rx_ctx_cnt = hints->rx_ctx_cnt;
+	if (user_modes & FI_MSG_PREFIX)
+		hints->msg_prefix_size = attr->msg_prefix_size;
```

Comment:
I *think* this is the flow before we reach this spot:
1. If the provider requires FI_MSG_PREFIX, the hints must also.  That check should have already been done.
1a. Duping the provider's info structure will have set msg_prefix_size
2. If the provider _may_ use FI_MSG_PREFIX, hints->mode is not verified for this bit.
2a. Duping the provider's info will have msg_prefix_size set.
2b. We need to clear msg_prefix_size here if hints->mode does NOT have FI_MSG_PREFIX

That is, I believe a provider that requires FI_MSG_PREFIX should work with the existing code.  For a provider that _wants_ but does not require FI_MSG_PREFIX, it should NOT set FI_MSG_PREFIX as a mode bit, but should set msg_prefix_size.  In this case, I think we need something like this:
```
if (!(user_mode &FI_MSG_PREFIX))
    attr->msg_prefix_size = 0;
```
We need to check the entire source flow to confirm that this is what would happen.